src/cluster.c

```diff
@@ -5913,10 +5954,15 @@ void clusterRedirectClient(client *c, clusterNode *n, int hashslot, int error_co
     } else if (error_code == CLUSTER_REDIR_MOVED ||
                error_code == CLUSTER_REDIR_ASK)
     {
+        /* Redirect to IP:port. Include plaintext port if cluster is TLS but
+         * client is non-TLS. */
+        int use_pport = (server.tls_cluster &&
```

Comment:
Why are we using server.tls_cluster? The clusterbus can not be using TLS but you can still have inbound TLS clients.

Reply:
If the cluster bus is non-TLS, the `port` in the cluster bus is always the non-TLS client port. That's legacy.

If we want to serve TLS ports to TLS clients in a non-TLS cluster (meaning a cluster with a non-TLS cluster bus), we can rename pport to "secondary port" (or port2) and send/gossip the TLS port in this way. (Future possibility.)

Reply:
Okay, we talked about this in the other comment and I understand the legacy behavior now.

I don't think the tls_cluster config should be the decider for what port we should show to end users. I guess the only time we can break that is in Redis 7. So maybe we can put a pin in that.